### PR TITLE
CI: Setup build tests + auto release CI

### PR DESCRIPTION
**Pull request by BitsOfAByte**
Discussed this PR with @chirpxiv before moving forward. This PR aims to streamline the PR testing & release process for Ktisis with the help of GitHub Actions.

An example release using this CI can be found here:
https://github.com/BitsOfAByte/container-testing-ci/releases/tag/v0.7.2.3

An example build run can be found here:
https://github.com/BitsOfAByte/container-testing-ci/actions/runs/3920817168/jobs/6702655988

An example repo.json update can be found here (it uses the current repo, ignore the url change)
https://github.com/BitsOfAByte/container-testing-ci/commit/9c868eb7bb6f7c21a54f9aed9dbf387742e18451

### CODE REVIEW

**Comment by chirpxiv** [comment]
no way I cant believe Blooym is already releasing Ktisis 0.7 for us

File: Ktisis/Ktisis.csproj [hunk 1]
```diff
@@ -24,6 +24,10 @@
   <PropertyGroup>
     <DalamudLibPath>$(appdata)\XIVLauncher\addon\Hooks\dev\</DalamudLibPath>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(OS)' != 'Windows_NT' OR '$(CI)' == 'true'">
```
Comment (Fayti1703) [inline Ktisis/Ktisis.csproj:27]: I know I'm a bit late on this, but shouldn't this condition just be `$(CI) == 'true'`?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        run: dotnet build -c Debug --no-restore --nologo
+
+      - name: Build plugin in release mode
+        run: dotnet build -c Release --no-restore --nologo
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: Build Artifacts
+          path: Ktisis/bin/
```
Comment (Fayti1703) [inline .github/workflows/test.yml:62]: Should a test build really upload artifacts?

Comment (Fayti1703) [reply .github/workflows/test.yml:62]: Having looked a bit more into it, this is fine. (It only drops it into the workflow results)

File: Ktisis/Ktisis.csproj (hunk 1 above)
Comment (BitsOfAByte) [reply Ktisis/Ktisis.csproj:27]: It's a holdover from another project that allows me to use a VSCode devcontainer and set the same environment variable for Dalamud location, doesn't need to stay if unwanted

**Pull request merged**